src/chapters/git/exercises.rst

```diff
@@ -134,6 +134,72 @@ a different service to create and share remote repositories.
+#. Click the *Sign Up* button and follow the on-screen directions.
+#. Create a *Personal Access Token* (see below).
+#. Once you have an account and a personal access token, you are ready to store
+   your remote work!
+
+Create a Personal Access Token (PAT)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. index:: personal access token
+
+To push and pull from GitHub, users must create a **personal access token**. A
+PAT takes the place of a password, and the token process is considered more
+secure than a username/password verification.
+
+After you create your PAT, you will use it anytime you interact with GitHub
+from the terminal.
+
+.. admonition:: Example
+
+   In the terminal, interactions between your computer and GitHub might look
+   something like this:
+
+   .. sourcecode:: bash
+
+      $ git push origin main
+      Username: your_username
+      Password: your_token
+
+   The bottom two lines will prompt you to enter your GitHub username and
+   personal access token.
+
+Some users question the need for a PAT, since it looks like another password
+they have to remember. Rather than diving into a long debate and explanation,
+we'll focus on the main point: *GitHub requires a PAT or similar token*. The
+service is incredibly helpful, and we want to use it, so we'll follow the
+developers' advice.
+
+Creating a PAT for your account is fairly straightforward. GitHub provides
+detailed instructions, so we will use their words. Login to your account, then
+carefully follow steps 1 - 9 for
+`Creating a Token <https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-token>`__.
+
+**Tips**:
+
+#. The checkboxes in step 7 select what you're allowed to do from the terminal.
+   For now, just choose the *repo* option.
+
+   There's no harm in selecting more options, but you won't need any of them
+   for this course.
+#. After you generate your PAT in step 8, *copy and save it somewhere safe*!
+   Your new PAT will NOT be an easy-to-remember sequence of characters (that's
+   the whole point). Record it somewhere.
+
+   If you use a password manager, that's a perfect place to keep your PAT.
+   If you use an unsecured spreadsheet or a folded piece of paper, you want to
+   break that habit now.
+#. If you expect to frequently push and pull from a repository during class,
+   you can save your PAT in memory for a short time. Run the command:
+
+   .. sourcecode:: bash
+
+      $ git config credential.helper 'cache --timeout=3600'
+
+   The next time you access your remote repo, Git will ask for your username
+   and PAT. It will then remember your credentials for a certain amount of
+   time. In the example above, ``timeout=3600`` saves your information for 1
+   hour (3600 seconds). You can adjust the amount of time up or down as needed.
+#. **Mac Users**: At the bottom of the PAT documentation page, you can find
```

Comment:
So are we suppose to not store it?  I set-up the new system but didn't give it a time limit?  Is it because it is my own personal computer that it is not necessary?  Is this included because it is a school computer? Or should I be regenerating a PAT more often or use instead of a SSH key?

Reply:
Good catch.  I removed the **Mac users** tip, since using the Keychain app isn't relevant for high school machines.

As for the cache limit, I tried to balance several competing issues:
1) The students will be using shared machines, so I'm not sure how setting up an SSH key would work in this scenario (plus, I found the instructions for setting up SSH a little more obtuse).
2) PATs are a thing now, so the kids have to use them.  However, it's a bear to type them every time we need to push/pull.  It's also not realistic to expect the kids to remember their PATs, nor is it an efficient to use class time to regenerate new ones each day.  This leads to caching the credentials.
3) Which, of course, runs headlong into the shared machine problem.  Indefinite credential storage... I'm not sure what happens when a student in the next class enters `git push`.  I don't think we can predict whether or not a student's credentials remain cached after they logout of the machine (and kids don't always logout).

Any suggestions for how to cleverly resolve this will be much appreciated!